### PR TITLE
DropDownList: Prevent unnecessary dataSource requests (T876423)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 1]
```diff
@@ -601,6 +601,19 @@ const DropDownList = DropDownEditor.inherit({
         return this._searchValue().toString().length >= this.option('minSearchLength');
     },
 
+    _needClearFilter: function() {
+        return !this._canKeepDataSource() ? this._needPassDataSourceToList() : false;
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_list.js]:
```suggestion
        return this._canKeepDataSource() ? false : this._needPassDataSourceToList();
```

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 2]
```diff
@@ -601,6 +601,19 @@ const DropDownList = DropDownEditor.inherit({
         return this._searchValue().toString().length >= this.option('minSearchLength');
     },
 
+    _needClearFilter: function() {
+        return this._canKeepDataSource() ? false : this._needPassDataSourceToList();
+    },
+
+    _canKeepDataSource: function() {
+        const isMinSearchLengthExceeded = this._isMinSearchLengthExceeded();
+        return this._dataSource?._isLoaded &&
```
Comment (San4es) [inline js/ui/drop_down_editor/ui.drop_down_list.js]: Use the public `isLoaded()` method instead.

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js [hunk 3]
```diff
@@ -1310,6 +1310,39 @@ QUnit.module('dataSource integration', moduleConfig, function() {
...
+            displayExpr: 'name',
+            searchExpr: 'name',
+            searchTimeout: 0,
+            minSearchLength: 3
+        });
+
+        const $input = $dropDownList.find('.' + TEXTEDITOR_INPUT_CLASS);
+        const kb = keyboardMock($input);
+
+        kb.type('1');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js]: `kb.type('123')` ?
or
```
kb
    .type('1')
    .type('2')
    .type('3')
```
?

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js [hunk 4]
```diff
@@ -1310,6 +1310,39 @@ QUnit.module('dataSource integration', moduleConfig, function() {
...
+        kb.type('2');
+        kb.type('3');
+
+        assert.strictEqual(loadStub.callCount, 2);
+
+        kb.press('backspace');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js]:
```
kb
    .press('backspace')
    .press('backspace')
```
?

**Review by San4es — APPROVED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

**Pull request merged**